### PR TITLE
MattT/APPEALS-8447: Adjust "Docket appeal" Modal for CSP

### PR DESCRIPTION
**Pull request by ThorntonMatthew**
Resolves [APPEALS-8447](https://vajira.max.gov/browse/APPEALS-8447)

### Description
1. Allows for different organization names to be inserted into the body of the "Docket appeal" modal depending on which workflow the appeal has been through.
2. Makes the Submit button blue with a hover effect.
3. Adds "Optional" label to textarea field on the "Docket appeal" modal.

### Acceptance Criteria
- [ ] The following occurs when the "Docket appeal" task action is selected
     - [ ] A modal appears to confirm intent 
     - [ ] Modal Header: Docket appeal
     - [ ] Modal informational text: Please confirm that the document provided by CSP are available in VBMS before docketing this appeal.
         - [ ] Once you confirm, the appeal will be established. Please remember to send the docketing letter out to all parties and representatives.
         - [ ] Provide instructions and context for this action:
     - [ ] BVA Intake user can provide additional context for the action
         - [ ] Field type: optional free text field
     - [ ] Modal contains the following selectable options:
         - [ ] Cancel
            - [ ] User is returned to case details screen and selection is not retained
         - [ ] Confirm
            - [ ] Select to execute task action
- [ ] The following occurs when the "Docket appeal" task action is executed 
   - [ ] The predocket task is completed
   - [ ] The appeal appears on the Completed tab of BVA Intake organization queue
   - [ ] BVA Intake user is returned to the BVA Intake organization queue 
      - [ ] A message appears confirming the appeal has been docketed

### Testing Plan
1. Log into Caseflow as an Intake user (ex: `BVAISHAW`). You will need to intake three appeals with the specified issues:

   - [ ] VHA CAMO
      - [ ] Benefit Type: Veterans Health Administration
      - [ ] Issue Category: Any except for the caregiver ones
      
   - [ ] VHA CSP
      - [ ] Benefit Type: Veterans Health Administration
      - [ ] Issue Category: Any caregiver issue type
          
   - [ ] Education
      - [ ] Benefit Type: Education
      - [ ] Predocket Needed?: Yes
      - [ ] Issue Category: Any is fine
      
I'd recommend using three different veteran file numbers for an easier time validating the appeals' presences in the BVA Intake queue.
  
2. Navigate to the Case Details pages for all three appeals (`/queue/appeals/<UUID>`)

3. Select "Docket appeal" from the task actions dropdown. Ensure that the modals match the [designs](https://bipuiux.invisionapp.com/console/share/P2MB68RZ73F/933825336/play).

4. On at least one modal, submit it without entering any text into the text field. Submit the other two.

5. Navigate to BVA Intake's completed tab (`/organizations/bva-intake?tab=bvaIntakeCompletedTab`) All three appeals should be here now!
   

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 WORKFLOW|BEFORE|AFTER
 ---|---|---
 VHA CAMO|pre-changes-docket|post-changes-vha-docket
 VHA CSP|pre-changes-docket|post-changes-csp-docket
 Education (EMO & RPO)|pre-changes-docket|docket-appeal-edu-service

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: spec/feature/queue/pre_docket_spec.rb [hunk 1]
```diff
@@ -837,5 +889,19 @@ def bva_intake_dockets_appeal
 
       expect(emo_task.appeal.tasks.last.assigned_to). to eq emo
     end
+
+    it "BVA Intake's 'Docket appeal' modal contains correct org name" do
```
Comment (ThorntonMatthewD) [inline spec/feature/queue/pre_docket_spec.rb:893]: Authenticate as BVA Intake user

File: spec/feature/queue/pre_docket_spec.rb [hunk 2]
```diff
@@ -72,12 +79,42 @@
...
+
+          find_link("#{veteran.name} (#{veteran.file_number})").click
+
+          click_dropdown(text: Constants.TASK_ACTIONS.DOCKET_APPEAL.label)
+
+          expect(page).to have_content(
+            format(COPY::DOCKET_APPEAL_MODAL_BODY, "CSP")
+          )
+
+          find("button", class: "usa-button", text: "Submit").click
```
Comment (ThorntonMatthewD) [inline spec/feature/queue/pre_docket_spec.rb:110]: Button text should be "Confirm"

File: app/repositories/task_action_repository.rb [hunk 3]
```diff
@@ -458,10 +458,21 @@ def review_decision_draft(task, user)
...
+      most_recent_child_task = task.children.first
+
+      # The last organization to work the appeal before sending it to BVA Intake
+      # for docketing.
+      pre_docket_org = case most_recent_child_task.assigned_to
+                       when VhaCamo.singleton then "VHA"
+                       when VhaCaregiverSupport.singleton then "CSP"
+                       else
+                         "Education"
+                       end
```
Comment (ajspotts) [inline app/repositories/task_action_repository.rb:471]: 🙌 Nice! Wondering if we want to potentially consider moving these references into the COPY.json file, might be overkill though

**Review by ajspotts — APPROVED** [review]
lgtm!

**Pull request merged**